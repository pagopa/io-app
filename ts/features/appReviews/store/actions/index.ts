import { ActionType, createStandardAction } from "typesafe-actions";
import { AppFeedbackUri } from "../../../../../definitions/content/AppFeedbackUri";

export type TopicKeys = keyof AppFeedbackUri;

export const appReviewPositiveFeedback = createStandardAction(
  "APP_REVIEW_POSITIVE_FEEDBACK"
)();

export const appReviewNegativeFeedback = createStandardAction(
  "APP_REVIEW_NEGATIVE_FEEDBACK"
)<TopicKeys>();

// Reset feedback data just for testing purposes
export const clearFeedbackDatas = createStandardAction(
  "APP_REVIEW_RESET_FEEDBACK"
)();

export type AppFeedbackActions =
  | ActionType<typeof appReviewPositiveFeedback>
  | ActionType<typeof appReviewNegativeFeedback>
  | ActionType<typeof clearFeedbackDatas>;
