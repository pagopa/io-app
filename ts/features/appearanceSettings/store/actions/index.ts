/**
 * Action types and action creator related to the Appearance settings.
 */
import { ActionType, createStandardAction } from "typesafe-actions";

export const setShowAppearanceSettingsBanner = createStandardAction(
  "SHOW_APPEARANCE_SETTINGS_BANNER"
)<boolean>();

export type AppearanceSettingsActions = ActionType<
  typeof setShowAppearanceSettingsBanner
>;
