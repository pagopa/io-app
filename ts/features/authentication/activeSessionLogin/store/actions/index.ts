import { ActionType, createStandardAction } from "typesafe-actions";
import { SpidIdp } from "../../../../../utils/idps";
import { SessionToken } from "../../../../../types/SessionToken";

export const setActiveSessionLoginLocalFlag = createStandardAction(
  "SET_ACTIVE_SESSION_LOGIN_LOCAL_FLAG"
)<boolean>();

export const setStartActiveSessionLogin = createStandardAction(
  "SET_START_ACTIVE_SESSION_LOGIN"
)();

export const setRetryActiveSessionLogin = createStandardAction(
  "SET_RETRY_ACTIVE_SESSION_LOGIN"
)();

export const setIdpSelectedActiveSessionLogin = createStandardAction(
  "SET_IDP_SELECTED_ACTIVE_SESSION_LOGIN"
)<SpidIdp>();

export const activeSessionLoginFailure = createStandardAction(
  "ACTIVE_SESSION_LOGIN_FAILURE"
)();

export const activeSessionLoginSuccess = createStandardAction(
  "ACTIVE_SESSION_LOGIN_SUCCESS"
)<SessionToken>();

export const setFastLoginOptSessionLogin = createStandardAction(
  "SET_FAST_LOGIN_OPT_IN_ACTIVE_SESSION_LOGIN"
)<boolean>();

export const consolidateActiveSessionLoginData = createStandardAction(
  "CONSOLIDATE_ACTIVE_SESSION_LOGIN_DATA"
)<{
  token: SessionToken;
  idp: SpidIdp;
  fastLoginOptIn: boolean;
}>();

export const setFinishedActiveSessionLoginFlow = createStandardAction(
  "SET_FINISHED_ACTIVE_SESSION_LOGIN_FLOW"
)();

export const setLoggedOutUserWithDifferentCF = createStandardAction(
  "SET_LOGGED_OUT_USER_WITH_DIFFERENT_CF"
)();

export type LoginInfoActions =
  | ActionType<typeof setActiveSessionLoginLocalFlag>
  | ActionType<typeof setStartActiveSessionLogin>
  | ActionType<typeof setRetryActiveSessionLogin>
  | ActionType<typeof setIdpSelectedActiveSessionLogin>
  | ActionType<typeof activeSessionLoginFailure>
  | ActionType<typeof activeSessionLoginSuccess>
  | ActionType<typeof consolidateActiveSessionLoginData>
  | ActionType<typeof setFastLoginOptSessionLogin>
  | ActionType<typeof setFinishedActiveSessionLoginFlow>
  | ActionType<typeof setLoggedOutUserWithDifferentCF>;
