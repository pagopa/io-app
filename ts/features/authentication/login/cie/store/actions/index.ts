export * from "./cie";
export * from "./cieLogin";
