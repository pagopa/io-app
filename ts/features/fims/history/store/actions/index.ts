import {
  ActionType,
  createAsyncAction,
  createStandardAction
} from "typesafe-actions";
import { ConsentsResponseDTO } from "../../../../../../definitions/fims_history/ConsentsResponseDTO";
import { FimsExportSuccessStates } from "../reducer";

export type FimsHistoryGetPayloadType = {
  shouldReloadFromScratch?: boolean;
  continuationToken?: string;
};

export const fimsHistoryGet = createAsyncAction(
  "FIMS_GET_HISTORY_REQUEST",
  "FIMS_GET_HISTORY_SUCCESS",
  "FIMS_GET_HISTORY_FAILURE"
)<FimsHistoryGetPayloadType, ConsentsResponseDTO, string>();

export const fimsHistoryExport = createAsyncAction(
  "FIMS_HISTORY_EXPORT_REQUEST",
  "FIMS_HISTORY_EXPORT_SUCCESS",
  "FIMS_HISTORY_EXPORT_FAILURE"
)<void, FimsExportSuccessStates, void>();

export const resetFimsHistoryState = createStandardAction(
  "RESET_FIMS_HISTORY_STATE"
)<void>();

export const resetFimsHistoryExportState =
  createStandardAction("RESET_FIMS_HISTORY")<void>();

export type FimsHistoryActions =
  | ActionType<typeof fimsHistoryGet>
  | ActionType<typeof fimsHistoryExport>
  | ActionType<typeof resetFimsHistoryExportState>
  | ActionType<typeof resetFimsHistoryState>;
