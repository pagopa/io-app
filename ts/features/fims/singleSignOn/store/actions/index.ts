import {
  ActionType,
  createAsyncAction,
  createStandardAction
} from "typesafe-actions";
import { ConsentData } from "../../types";
import { FimsErrorStateType } from "../reducers";
import { ServiceId } from "../../../../../../definitions/backend/ServiceId";

type FimsGetConsentsListRequestType = {
  ctaText: string;
  ctaUrl: string;
};

type FimsGetRedirectUrlAndOpenIABRequestType = {
  acceptUrl?: string;
  serviceId: ServiceId;
};

export const fimsGetConsentsListAction = createAsyncAction(
  "FIMS_GET_CONSENTS_LIST_REQUEST",
  "FIMS_GET_CONSENTS_LIST_SUCCESS",
  "FIMS_GET_CONSENTS_LIST_FAILURE"
)<FimsGetConsentsListRequestType, ConsentData, FimsErrorStateType>();

// note: IAB==InAppBrowser
export const fimsGetRedirectUrlAndOpenIABAction = createAsyncAction(
  "FIMS_GET_REDIRECT_URL_REQUEST",
  "FIMS_GET_REDIRECT_URL_SUCCESS",
  "FIMS_GET_REDIRECT_URL_FAILURE"
)<FimsGetRedirectUrlAndOpenIABRequestType, void, FimsErrorStateType>();

export const fimsCancelOrAbortAction = createStandardAction(
  "FIMS_CANCEL_OR_ABORT"
)<void>();

export type FimsSSOActions =
  | ActionType<typeof fimsGetConsentsListAction>
  | ActionType<typeof fimsGetRedirectUrlAndOpenIABAction>
  | ActionType<typeof fimsCancelOrAbortAction>;
