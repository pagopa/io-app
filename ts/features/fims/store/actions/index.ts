import { ActionType, createAsyncAction } from "typesafe-actions";
import { ConsentData } from "../../types";

type FimsGetConsentsListRequestType = {
  ctaUrl: string;
};

type FimsGetRedirectUrlAndOpenIABRequestType = {
  acceptUrl?: string;
};

export const fimsGetConsentsListAction = createAsyncAction(
  "FIMS_GET_CONSENTS_LIST_REQUEST",
  "FIMS_GET_CONSENTS_LIST_SUCCESS",
  "FIMS_GET_CONSENTS_LIST_FAILURE"
)<FimsGetConsentsListRequestType, ConsentData, string>();

// note: IAB==InAppBrowser
export const fimsGetRedirectUrlAndOpenIABAction = createAsyncAction(
  "FIMS_GET_REDIRECT_URL_REQUEST",
  "FIMS_GET_REDIRECT_URL_SUCCESS",
  "FIMS_GET_REDIRECT_URL_FAILURE"
)<FimsGetRedirectUrlAndOpenIABRequestType, void, string>();

export type FimsActions =
  | ActionType<typeof fimsGetConsentsListAction>
  | ActionType<typeof fimsGetRedirectUrlAndOpenIABAction>;
