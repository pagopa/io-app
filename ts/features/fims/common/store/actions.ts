import { FimsSSOActions } from "../../singleSignOn/store/actions";

export type FimsActions = FimsSSOActions;
