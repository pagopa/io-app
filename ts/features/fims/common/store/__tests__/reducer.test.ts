import { applicationChangeState } from "../../../../../store/actions/application";
import fimsReducer from "../reducer";

describe("fimsReducer", () => {
  it("should match snapshot", () => {
    const fimsState = fimsReducer(undefined, applicationChangeState("active"));
    expect(fimsState).toMatchSnapshot();
  });
});
