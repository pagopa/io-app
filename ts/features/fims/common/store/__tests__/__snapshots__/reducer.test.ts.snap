// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fimsReducer should match snapshot 1`] = `
{
  "history": {
    "consentsList": {
      "kind": "PotNone",
    },
    "historyExportState": {
      "kind": "undefined",
    },
  },
  "sso": {
    "ctaText": undefined,
    "currentFlowState": "idle",
    "relyingPartyServiceId": undefined,
    "relyingPartyUrl": undefined,
    "ssoData": {
      "kind": "PotNone",
    },
  },
}
`;
